discovery-provider/src/models/models.py

```diff
@@ -29,6 +30,24 @@
 Base: Any = declarative_base()
 logger = logging.getLogger(__name__)
 
+# Listen for instrumentation of attributes on the base class
+# to add a listener on that attribute whenever it is set
+@event.listens_for(Base, "attribute_instrument")
+def configure_listener(class_, key_, inst):
+    # Check that the attribute is a column (we only validate columns)
+    if not hasattr(inst.property, "columns"):
+        return
+
+    # Listen for set events on the attribute to run our default validations
+    @event.listens_for(inst, "set", retval=True)
+    def set_(target, value, oldvalue, initiator):
+        column_type = getattr(target.__class__, inst.key).type
+        if type(column_type) in (String, Text) and value and isinstance(value, str):
```

Comment:
wonder if we should handle Unicode and UnicodeText here just in case someone adds that and doesn't check back here.

Reply:
Good idea